layers/CMakeLists.txt

```diff
@@ -15,11 +15,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ~~~
+set(TARGET_NAMES VkLayer_khronos_timeline_semaphore VkLayer_khronos_synchronization2 VkLayer_khronos_shader_object)
 
-if(WIN32)
-    set(TARGET_NAMES VkLayer_khronos_synchronization2 VkLayer_khronos_shader_object) # timeline_semaphore doesn't build on windows yet
-else()
-    set(TARGET_NAMES VkLayer_khronos_timeline_semaphore VkLayer_khronos_synchronization2 VkLayer_khronos_shader_object)
+# timeline_semaphore doesn't build on windows yet
+if (WIN32)
+    list(REMOVE_ITEM ${TARGET_NAMES} "VkLayer_khronos_timeline_semaphore")
+endif()
+
+if (CMAKE_SIZEOF_VOID_P EQUAL "4")
```

Comment:
This is unusual, so a comment saying that the Vulkan SDK has deliberately chosen not to ship 32-bit `VkLayer_khronos_shader_object` files would save confusion in the future.